bo/marktteilnehmer_test.go

```diff
@@ -0,0 +1,119 @@
+package bo
+
+import (
+	"encoding/json"
+	"github.com/corbym/gocrest/is"
+	"github.com/corbym/gocrest/then"
+	"github.com/go-playground/validator/v10"
+	"github.com/hochfrequenz/go-bo4e/com"
+	"github.com/hochfrequenz/go-bo4e/enum/anrede"
+	"github.com/hochfrequenz/go-bo4e/enum/botyp"
+	"github.com/hochfrequenz/go-bo4e/enum/geschaeftspartnerrolle"
+	"github.com/hochfrequenz/go-bo4e/enum/landescode"
+	"github.com/hochfrequenz/go-bo4e/enum/marktrolle"
+	"github.com/hochfrequenz/go-bo4e/enum/rollencodetyp"
+	"strings"
+)
+
+// Test_Marktteilnehmer_Deserialization deserializes an Marktteilnehmer json
+func (s *Suite) Test_Marktteilnehmer_Deserialization() {
+	var mt = Marktteilnehmer{
+		Marktrolle:       marktrolle.LF,
+		Makoadresse:      "edifact@my-favourite-marketpartner.de",
+		Rollencodetyp:    rollencodetyp.DVGW,
+		Rollencodenummer: "9903100000006",
+		Geschaeftspartner: Geschaeftspartner{
+			BusinessObject: BusinessObject{
+				BoTyp:             botyp.Geschaeftspartner,
+				VersionStruktur:   "1",
+				ExterneReferenzen: nil,
+			},
+			Anrede:               anrede.Divers,
+			Name1:                "Müller",
+			Name2:                "Lieschen",
+			Name3:                "",
+			Gewerbekennzeichnung: false,
+			HrNummer:             "handelsregister foo",
+			Amtsgericht:          "amtsgericht bar",
+			Kontaktweg:           0,
```

Comment:
Ah das enum wird hier (ähnlich wie bei python) auch über zahlen definiert.

Reply:
ja genau. wobei 0 wegen des `iota+1`  in der definition des enums immer sowas wie undefined ist. So können wir dann Validieren, dass wert!=0 de facto heißt, dass ein sinnvoller Wert angegeben ist.